app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadService.java

```diff
@@ -404,6 +412,8 @@ private void updateForeground() {
 
             input = response.body().source();
 
+            Log.d("kiwixdownloadservice", "Got valid chunk");
```

Comment:
How about moving this log statement after the next line that gets the lastTime and adding that time to the log message? that way we could read the log and calculate the timings of a partial download and see when it failed (where the time would be formatted for both machine-based processing and human readability i.e. 2 values). e.g. something like Log.d(..., "Got valid chunk at: ", lastTime, human_readable_time(lastTime));